src/main/java/com/almightyalpaca/discord/jdabutler/commands/Dispatcher.java

```diff
@@ -0,0 +1,119 @@
+package com.almightyalpaca.discord.jdabutler.commands;
+
+import com.almightyalpaca.discord.jdabutler.Bot;
+import com.almightyalpaca.discord.jdabutler.commands.commands.*;
+import com.almightyalpaca.discord.jdabutler.util.StringUtils;
+import net.dv8tion.jda.core.entities.TextChannel;
+import net.dv8tion.jda.core.entities.User;
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Dispatcher extends ListenerAdapter {
+
+    private Map<String, Command> commands = new ConcurrentHashMap<>();
+    private ExecutorService pool = Executors.newCachedThreadPool(); // next pr: remove all of these indents
+
+    {
+        registerCommand(new HelpCommand());
+        registerCommand(new VersionsCommand());
+        registerCommand(new Eval());
+        registerCommand(new DocsCommand());
+        registerCommand(new Gradle());
+        registerCommand(new Maven());
+        registerCommand(new Jar());
+        registerCommand(new BuildGradle());
+        registerCommand(new Notify());
+        registerCommand(new Ping());
+        registerCommand(new Uptime());
+        registerCommand(new Changelog());
+    }
+
+    public boolean registerCommand(Command command) {
+        if (command.getName().contains(" ")) {
+            throw new IllegalArgumentException("Name must not have spaces!");
+        }
+        if (commands.containsKey(command.getName().toLowerCase())) {
```

Comment:
The map will not have duplicate keys when you put one, it will override existing values associated with that key.